File: Python3/Hashmap & Set/1207. Unique Number of Occurrences.py
""" 1207. Unique Number of Occurrences
Difficulty: Easy
Submission Link: https://leetcode.com/problems/unique-number-of-occurrences/submissions/1148976975/
 """

class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        frequency = {}
        for num in arr:
            frequency[num] = frequency.get(num,0) + 1
        return len(frequency) == len(set(frequency.values()))

File: C++/Hashmap & Set/1207. Unique Number of Occurrences.cpp
/* 1207. Unique Number of Occurrences
Difficulty: Easy
Submission Link: https://leetcode.com/problems/unique-number-of-occurrences/submissions/1148974551/ */

class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> frequency;
        for(int num : arr) 
            frequency[num]++;
        unordered_set<int> storage;
        for(auto key : frequency)
            storage.insert(key.second);
        return frequency.size() == storage.size();
    }
};